persist-cli/src/main/java/io/ballerina/persist/cmd/Push.java

```diff
@@ -115,73 +117,105 @@ public void execute() {
             return;
         }
 
-        Path projectPath = Paths.get(this.sourcePath).toAbsolutePath();
-        Path persistTomlPath = Paths.get(this.sourcePath, PERSIST_DIRECTORY, PERSIST_TOML_FILE);
-
-        if (!Files.exists(persistTomlPath)) {
-            errStream.println("Persist project is not initiated. Please run `bal persist init` " +
-                    "to initiate the project before the database schema generation. ");
+        Path persistDir = Paths.get(this.sourcePath, PERSIST_DIRECTORY);
+        if (!Files.isDirectory(persistDir, NOFOLLOW_LINKS)) {
+            errStream.println("The persist directory inside the Ballerina project doesn't exist. " +
+                    "Please run `bal persist init` to initiate the project before generation");
             return;
         }
 
-        try {
-            BuildProject buildProject = validateSchemaFile(projectPath, false);
-            io.ballerina.projects.Module module = getEntityModule(buildProject);
-            Module entityModule = BalProjectUtils.getEntities(module);
-            ArrayList<Entity> entityArray = new ArrayList<>(entityModule.getEntityMap().values());
-            String[] sqlScripts = SqlScriptGenerationUtils.generateSqlScript(entityArray);
-            SqlScriptGenerationUtils.writeScriptFile(sqlScripts,
-                    Paths.get(projectPath.toString(), PERSIST_DIRECTORY));
-        } catch (ProjectException | BalException e) {
-            errStream.println("Error occurred while generating SQL schema. " + e.getMessage());
+        List<Path> schemaFilePaths;
+        try (Stream<Path> stream = Files.list(persistDir)) {
+            schemaFilePaths = stream.filter(file -> !Files.isDirectory(file))
+                    .filter(file -> file.toString().toLowerCase(Locale.ENGLISH).endsWith(".bal"))
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            errStream.println("Error while listing the persist schema files in persist directory. " + e.getMessage());
```

Comment:
```suggestion
            errStream.println("Error while listing the persist model definition files in persist directory. " + e.getMessage());
```
Shall we use "model definition" instad of schema? schem applies mostly for database related stuff